hphp/runtime/vm/jit/smashable-instr-arm.cpp

```diff
@@ -261,7 +273,25 @@ ConditionCode smashableJccCond(TCA inst) {
   if (br->Bits(31, 10) != 0x3587C0 || br->Bits(4, 0) != 0) return CC_NP;
 
   return arm::convertCC(InvertCondition(static_cast<Condition>(b->Bits(3, 0))));
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+TCA getSmashableFromTargetAddr(TCA addr) {
```

Comment:
Can you document this function (in the header file)?

I'm not totally clear on what it does... it looks like if you pass it an address which is the target of an immediately preceding smashable jmp, you get that smashable jmp returned.  That's bizarrely specific for such a general-sounding function name, but maybe I'm misunderstanding something.